### DISCUSSION TITLE
confused by cons

### DISCUSSION THREAD

**Discussion by DanielAjoy** [h3rald/min#173] [Bugs: Fixed]
I'm confused by this behavior of **cons**. It seems new items are added to the original empty list.

```
(() cons cons) ^list

$ "a" "B" list
 (
   "a"
   "B"
 )

$ "a" "B" list
 (
   "a"
   "B"
   "a"
   "B"
 )

$ "a" "B" list
 (
   "a"
   "B"
   "a"
   "B"
   "a"
   "B"
 )
```

That is not what happens in Joy

```
DEFINE
lista == [] cons cons .

2 3 lista .
[2 3]

2 3 lista .
[2 3]
```

**Comment by h3rald** [comment]
Ohhhh yes, this could actually be a rather nasty bug. For some reason the same list is re-used within the lambda. I will need to investigate this further when I have a chance. Of course a PR would be welcome if you/anyone figures out why this is happening.

**Comment by h3rald** [comment]
Moving to fixed as this was addressed.
